### ISSUE TITLE
Seeding Report: Test Cancel Seeding Log Edit

### ISSUE THREAD

**Issue by braughtg** [DickinsonCollege/FD2School-FarmData2#202]
The Seeding Report allows the values within a seeding log (i.e. a row) to be edited by clicking the “Edit” button (i.e. the blue pencil button).  When a row is being edited the “Edit” button changes to two buttons. The “Save” button (the green check mark) will save the changes to the database.  The “Cancel” button (the brown X) discards any edits and does not change the database.  

**Notes:**
  - The test must enter data into the form and use the “Submit” button
  - The test should not use the Seeding Report to check that the database was not changed.
  - The test must use appropriate FarmOSAPI functions to:
    - check that the database was not changed.
  - If you find that you have scrambled the database you can always reset to the default sample database running the command `setDB sample` in a Terminal in the development environment.

**Resources:**
  - The examples in the dbtest sub-tab of the FD2 Example tab will be helpful.  These should be studied and understood before beginning coding for this issue.  
  - The source for the examples can be found in `farmdat2/farmdata2_modules/fd2_example/dbtest`.

**Comment by Shahir-47** [comment]
I would like to work on this!

**Comment by JinLeeGG** [comment]
I would like to work on this!

**Comment by won369369** [comment]
I would like to work on this!

**Comment by Shahir-47** [comment]
Hi @braughtg and @johnmaccormick, the seeding report does not show the edit button if the window width or height is less than 900 pixels. So it doesn't show the edit button on my laptop.

![image](https://user-images.githubusercontent.com/98346408/235825115-41d8c51b-2b78-4a2b-9842-cfda1c187cf0.png)


Here's the code for the custom table: 
`<custom-table data-cy="report-table" :columns="tableColumns" :rows='tableRows' csv-name="seedingReport_" @row-deleted="deleteRowLog" @row-edited='updateRow' @edit-clicked='disableFilters' @edit-canceled='enableFilters' :can-edit='!isMobile' :can-delete='!isMobile'></custom-table>`

You can see that the property 'isMobile' needs to be false in order for the prop canEdit to be true. The if condition given below tells that the width and height needs to be greater than or equal to 900 pixels which isn't possible on my device. 

` if (window.screen.width < 900 || window.screen.height < 900) {
this.isMobile = true
                }`

When I change the pixels in the if condition to 500 pixels like shown below:

`if (window.screen.width < 500 || window.screen.height < 500) {
this.isMobile = true
                }`

The edit button is now clearly visible:

![image](https://user-images.githubusercontent.com/98346408/235825294-d0770f59-9fa8-4a47-9d07-e512e542dc47.png)

For now, I've changed the if condition in the .html file so I can do cypress tests on my laptop but will change it back to how it was previously.

**Comment by Shahir-47** [comment]
I found another issue with the seeding report @braughtg @johnmaccormick,

When I press the edit button and select the drop-down menu for the crop, the crop names are repeated twice. For example, the crop name BEET is repeated twice in the drop-down menu  as you can see in the screenshots below:

![image](https://user-images.githubusercontent.com/98346408/235825867-f109da40-ca7f-476a-a220-46435f6102cf.png)
![image](https://user-images.githubusercontent.com/98346408/235825973-df7ce95c-7dd5-4c0d-89a1-a8829f4cdb2c.png)

I also used Vue to check the custom table component and found that the second column has a dropdown option with an array that has 222 crop names. I remember from our previous assignment that there were only 111 crop names.

![image](https://user-images.githubusercontent.com/98346408/235826368-7080a63d-6213-49b7-be57-0e05b3444213.png)

In that array you can clearly see down below that the element 0 and element 111 are clearly repeated, 

![image](https://user-images.githubusercontent.com/98346408/235826636-7004c6a7-c3d6-42f8-a7d0-7772765389f3.png)
 
![image](https://user-images.githubusercontent.com/98346408/235826737-13036da1-8b49-4ee2-be9e-77aa29fc767e.png)

Because of these duplications, my cypress tests fail.

**Comment by braughtg** [comment]
> the seeding report does not show the edit button if the window width or height is less than 900 pixels. So it doesn't show the edit button on my laptop.

Thanks for pointing this out and adding additional information about the location of the issue and a work around.  This seems to be the same issue reported in #230.  

I have seen this issue as well and it has also been reported on screens that render widths > 900 pixels.  So, I suspect that your laptop probably does allow rendering of widths greater than 900 pixels, suggesting that this issue is being caused by the way that the browser is reporting widths rather than the 900 number specifically.  That said, the workaround you suggest works well for testing at this point.

When you finish the tests, I hope you will consider working on a full solution to #230.

**Comment by braughtg** [comment]
> When I press the edit button and select the drop-down menu for the crop, the crop names are repeated twice.

Good catch!  @Shahir-47, would you be willing to create an issue for this bug in the Upstream FarmData2 issue tracker ()?  That would be super helpful in getting it fixed in the live project.

Once you post the issue, I hope your team will consider working on fixing it!

There was a similar issue with the Areas dropdown in the SeedingInput form (DickinsonCollege/FarmData2#460) which was fixed by PR DickinsonCollege/FarmData2#461.  There may be some useful insights there.

**Comment by Shahir-47** [comment]
Hi @braughtg, I created a new issue #235 for the duplicate crop names in the dropdown menu.
